build/configs/rtl8721csm/rtl8721csm_download.sh

```diff
@@ -44,17 +44,33 @@ CUR_PATH=$(dirname "$CURDIR")
 TOOL_PATH=${CUR_PATH}/../../tools/amebad
 IMG_TOOL_PATH=${TOOL_PATH}/image_tool
 BIN_PATH=${CUR_PATH}/../../output/bin
+TOP_PATH=${CUR_PATH}/../../..
+OS_PATH=${TOP_PATH}/os
 SUDO=sudo
 
+CONFIG=${OS_PATH}/.config
+source ${CONFIG}
+
 cp -p ${BIN_PATH}/km0_boot_all.bin ${IMG_TOOL_PATH}/km0_boot_all.bin
 cp -p ${BIN_PATH}/km4_boot_all.bin ${IMG_TOOL_PATH}/km4_boot_all.bin
 cp -p ${BIN_PATH}/km0_km4_image2.bin ${IMG_TOOL_PATH}/km0_km4_image2.bin
+if [[ "${CONFIG_APP_BINARY_SEPARATION}" == "y" ]];then
```

Comment:
smartfs image can be made anytime, not only for binary separation.
I think we should consider another way for checking.

Reply:
But, since we are only flashing to device for loadable configuration, I feel this condition should be okay.
@sunghan-chang what is your opinion ?

Reply:
As @jeongchanKim said, fs image could be used with flat build as well. That's why ARTIK05x support downloading fs image even it does not support loadable. FS is mandatory for laodable but loadable is not mandatory for fs. Sorry @thapav , but I agree @jeongchanKim 's opinion.

Reply:
Sure, in that case I will have to ask for new requirement from realtek. Let me do that & get back to you.

Reply:
This will be addressed once I receive image from realtek.

Reply:
You can now downlaod smartfs binary using **make download smartfs** command